### ISSUE TITLE
Errors on some queryables requests 

### ISSUE THREAD

**Issue by justb4** [geopython/pygeoapi#420]
**Description**
Some Collections give error when requesting `queryables`.

**Steps to Reproduce**
* go to https://demo.pygeoapi.io and select `master`/`stable` or `covid-19` endpoint
* select `queryables` of a `Collection`
* for example https://demo.pygeoapi.io/master/collections/dutch_windmills/queryables
* see `Internal Server Error`
* not all Collections give this error, e.g. https://demo.pygeoapi.io/master/collections/obs/queryables shows fields

**Expected behavior**
List of queryable fields or empty.

**Screenshots/Tracebacks**
See e.g. https://demo.pygeoapi.io/master/collections/dutch_windmills/queryables
gives:
`Internal Server Error The server encountered an internal error and was unable to complete your request. Either the server is overloaded or there is an error in the application.`

for all endpoints using `master` version. 
In some cases also error log output (e.g. `docker logs --follow pygeoapi_covid-19`) is shown, like on
https://demo.pygeoapi.io/covid-19/collections/cases/queryables

```
[2020-04-25T10:43:45Z] {/usr/local/lib/python3.8/dist-packages/flask/app.py:1891} ERROR - Exception on /collections/cases/queryables [GET]
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 2447, in wsgi_app
    response = self.full_dispatch_request()
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1952, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/usr/local/lib/python3.8/dist-packages/flask_cors/extension.py", line 161, in wrapped_function
    return cors_after_request(app.make_response(f(*args, **kwargs)))
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1821, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/usr/local/lib/python3.8/dist-packages/flask/_compat.py", line 39, in reraise
    raise value
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1950, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1936, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/pygeoapi/pygeoapi/flask_app.py", line 181, in get_collection_queryables
    headers, status_code, content = api_.get_collection_queryables(
  File "/pygeoapi/pygeoapi/api.py", line 88, in inner
    return func(cls, headers_, format_, *args, **kwargs)
  File "/pygeoapi/pygeoapi/linked_data.py", line 54, in inner
    return func(*args, **kwargs)
  File "/pygeoapi/pygeoapi/api.py", line 518, in get_collection_queryables
    'type': v['type']
TypeError: string indices must be integers
```

**Environment**
 - OS: any
 - Python version: 3
 - pygeoapi version: `master`

**Additional context**
NA

**Comment by tomkralidis** [comment]
Reproduced.  Looks like some updates are needed to OGR, PostgreSQL, SQLite (cc @jorgejesus) and MongoDB (cc @timtuun) providers.

A provider's `get_fields` method should return a dict structured like:

```python
{
    'field1': {
        'type': 'string'
    },
    'field2: {
        'type': 'int'
    }
}
```
Examples:
- [Elasticsearch](https://github.com/geopython/pygeoapi/blob/master/pygeoapi/provider/elasticsearch_.py#L88)
- [native CSV](https://github.com/geopython/pygeoapi/blob/master/pygeoapi/provider/csv_.py#L58)
- [native GeoJSON](https://github.com/geopython/pygeoapi/blob/master/pygeoapi/provider/geojson.py#L70)

**Comment by tomkralidis** [comment]
Fixed via 11ee7053fc374b106225d84a3d547419eaf0e938 and 6916099315f77a53b12e3733b883e609f33a81e1
